core/src/main/scala/kafka/tools/MirrorMaker.scala

```diff
@@ -566,6 +566,16 @@ object MirrorMaker extends Logging with KafkaMetricsGroup {
     }
 
     override def cleanup() {
+      // This is needed to unblock zookeeper listener thread if zookeeper
+      // listener thread calls requestAndWaitForCommit() after MirrorMaker thread
+      // has already exited loop of consuming and producing messages.
+      this.synchronized {
+        iter = null
+        if (immediateCommitRequested) {
```

Comment:
I personally the currently solution may be slightly more readable because calling notifyCommit() will trigger another this.synchronized(). But both solutions work for me. I will change it if you prefer calling notifyCommit() here.

Yeah the comment for `requestAndWaitForCommit()` needs update as well.